### DISCUSSION TITLE
Remote BTCPay Point of Sale App

### DISCUSSION THREAD

**Discussion by pavlenex** [btcpayserver/btcpayserver#2662] [Ideas / Feature Requests]
## BTCPay Point Of Sale Remote app concept

If we think of BTCPay, we cannot assume that scaling means every single merchant runs a server. BTCPay is about decentralizing power, and allowing anyone to accept payments but also enabling people to host servers for others so that their community can accept Bitcoin without having to run a server.

Currently, the only way to use BTCPay Server in retail is our web-based Point of sale app. 

This document tries to lay out the foundation for what would be a remote, mobile, PoS app that would allow users to use BTCPay without ever having to access the server using Greenfield API.

Ideal scenario: A tech-savvy community member deploys btcpay and local merchants just need to install app on their phone in order to accept btc, without ever having to visit the hosts website.

### Onboarding

- Enter your BTCPay Server URL
- Creates account on BTCPay automatically
- Email confirmation via the app (if required)
- Creates the store
- Creates the wallet (multisig or key automatically held in cloud/phone storage)
- Crates a simple PoS with a custom amount for people to request payments

### Features

- Simple custom entry PoS
- Simple wallet to manage funds 
- Invoice export
- Refund
- Lightning Network focused (LDK?)

### Mockups
![Screenshot 2021-07-05 at 12 07 24](https://user-images.githubusercontent.com/36959754/124455204-94b3c700-dd89-11eb-89d8-3454addd6df8.png)

**Comment by woutersamaey** [comment]
I get the feeling this discussion addresses 2 things at once:
1. You're choosing to get a wallet using someone else's BTCPay Server because you know them and trust their node.
2. You're adding an app on top of that. In this case the PoS, but it could be a crowdfund or other new type of future app just as well?

**Reply by pavlenex** [reply]
1. Not really, you can generate a wallet locally or we can have a multisig.
2. App in a sense of a mobile native app, not the `app concept` we have on the server, this would allow people to operate from their mobile phone without ever having to go through the website itself.

**Reply by woutersamaey** [reply]
I'm trying to see the real-world use case for this. Having a PoS app on such a small screen does not seam very practical. Maybe for a shop with a small amount of products that could be easily tapped? Or would the app be able to scan bar codes to add them to the shopping cart? If the store has more than a few products, entering the products to the store will also take time on a small screen, unless you offer Excel import or something like that.

Another use case may be a delivery guy asking for payment upon delivery. He can show a QR code to be paid.

Curious what your thoughts are...

Just a suggestion: Instead of building a mobile app, we could turn this into a Google Flutter project and build the app + website from the same codebase. Both interfaces could benefit, but it's more effort.

**Reply by pavlenex** [reply]
Custom amount pos, where merchant just types in the amount themselves. 
> How much do I owe you? 25$, types in generates the invoice.

This is to scale to smaller shops that don't need anything complex.

**Reply by Kukks** [reply]
If it helps, I had built a proof of concept of how this could look using Vue + Cordova early last year before work started on Greenfield: https://github.com/Kukks/btcpayserver-ui

**Reply by Zaxounette** [reply]
> Having a PoS app on such a small screen does not seam very practical.

This is the Light version of the BTCPayServer POS app:

![chrome_0vvgFTeBPh](https://user-images.githubusercontent.com/51208677/124462677-3ccd8e00-dd92-11eb-984c-719a0b55552a.png)

Pages like this are perfectly fine for mobile and small tablets.
If a smartphone app shows something in the lines of this it won't be a problem.

The merchant can use his regular tools to scan code bars and such, and just enters the total to generate the invoice through BTCPay.

**Reply by Kukks** [reply]
I would recommend that if we go towards a native app, we build using C# + Xamarin + Blazor Mobile BIndings + Ionic. I spent a considerable amount of time creating a base framework on how to achieve that(See: https://github.com/Kukks/Blazor.Ionic and https://github.com/Kukks/Hara) which is now used in production for [Chaincase](https://chaincase.app/)

**Reply by woutersamaey** [reply]
Okay, I understand a lot better now. Great suggestions and thanks for showing.

**Comment by fvalentiner** [comment]
Hi @pavlenex! 

I want to build a project in MAUI and would like to contribute to BTCPay. 

I have experience in the retail space and building Xamarin apps.

Do you have any recommendation as to how to start?

**Reply by Kukks** [reply]
Hi there! Perfect. MAUI would be a good fit. MAUI Desktop /Mobile Blazor Bindings was my target. Here is a POC I made last year: https://github.com/Kukks/BlazingPay

I think I would drop [Blazor.Ionic](https://github.com/Kukks/Blazor.Ionic) and use something a bit simpler (I do not have time to maintain an entire UI framework bridge too :)), such as MudBlazor

**Reply by Kukks** [reply]
My ideal tech stack:
* MAUI for deploying on mobile
* Blazor to be able to stay in sync with the website tech and the UI decisions we take on btcpay web
* Mobile Blazor Bindings /[Blazor Desktop](https://github.com/dotnet/maui/wiki/Blazor-Desktop) to bridge Blazor and MAUI and have smooth performance
* [MudBlazor ](MudBlazor )for a UI framework to rapidly develop the UI
* [Fluxor ](https://github.com/mrpmorris/fluxor)to manage the UI and application states

**Reply by fvalentiner** [reply]
@Kukks thanks for the description. I've been getting familiar with your POC. I'll get beck to you soon with any questions I might have.

**Reply by pavlenex** [reply]
@fvalentiner let us know if there's something we can do to help. I think we probably may want to design some sort of a low fidelity wireframe.

**Reply by fvalentiner** [reply]
@pavlenex yes that would be helpful. I can start with that. Any preference on what tool to use for the wireframe?

**Reply by pavlenex** [reply]
I would say, you can leave that on to our design contributors, though you're free to brainstorm. You should download the Mattermost app and join chat.btcpayserver.org so we can all chat about the direction, this isn't a small step and there should be a consensus on the direction. We already have a rough idea, but needs to be fleshed out. Let's connect there?

**Comment by BenGWeeks** [comment]
Out of interest, any progress on this?

**Comment by pavlenex** [comment]
https://github.com/btcpayserver/app
